tests/karma/osu_common.spec.ts

```diff
@@ -16,53 +16,227 @@
-    expect(Lang.has('common.confirmation')).toBe(true);
-  });
+  describe('test locale file loaded in test runner', () => {
+    it('should be loaded', () => {
+      expect(Lang.has('common.confirmation')).toBe(true);
+    })
+  })
+
+  describe('test groupColour', () => {
+    const group: GroupJson = {
+      colour: '#abcdef',
+      has_listing: true,
+      has_playmodes: true,
+      id: 1,
+      identifier: 'abc',
+      is_probationary: false,
+      name: 'ABC',
+      short_name: 'abc',
+    }
+
+    it('should return CSS object with correct colour', () => {
+      expect(osu.groupColour(group)).toEqual({
+        '--group-colour': '#abcdef',
+      } as React.CSSProperties)
+    })
+
+    it('should return CSS object with initial value when colour is null', () => {
+      expect(osu.groupColour({ ...group, colour: null })).toEqual({
+        '--group-colour': 'initial',
+      } as React.CSSProperties)
+    })
+
+    it('should return CSS object with initial value when group is undefined', () => {
+      expect(osu.groupColour()).toEqual({
+        '--group-colour': 'initial',
+      } as React.CSSProperties)
+    })
+  })
+
+  describe('test jsonClone', () => {
+    it('should return same object', () => {
+      expect(osu.jsonClone({ test: '1234' })).toEqual({ test: '1234' });
+    })
+
+    it('should return null', () => {
+      expect(osu.jsonClone(null)).toBe(null);
+    })
+
+    it('should return null when object is undefined', () => {
+      expect(osu.jsonClone(undefined)).toBe(null);
+    })
+  })
+
+  describe('test linkify', () => {
+    const textWithLink = 'Please visit https://link.com';
+
+    it('should return correct anchor', () => {
+      expect(osu.linkify(textWithLink)).toBe(
+        `Please visit <a href="https://link.com" rel="nofollow noreferrer">link.com</a>`
+      )
+    })
+
+    it('should return correct anchor with target blank', () => {
+      expect(osu.linkify(textWithLink, true)).toBe(
+        `Please visit <a href="https://link.com" rel="nofollow noreferrer" target="_blank">link.com</a>`
+      )
+    })
+  })
+
+  describe('test formatBytes', () => {
+    it('should return same value in bytes', () => {
+      expect(osu.formatBytes(100)).toBe('100 B');
+    })
+
+    it('should return correct value in KB', () => {
+      expect(osu.formatBytes(1000)).toBe('1.00 KB');
+    })
+
+    it('should return correct value in MB', () => {
+      expect(osu.formatBytes(1000000)).toBe('1.00 MB');
+    })
+
+    it('should return correct value in KB without trailing zero', () => {
+      expect(osu.formatBytes(1000, 0)).toBe('1 KB');
+    })
+  })
+
+  describe('test formatNumber', () => {
+    it('should return null', () => {
+      expect(osu.formatNumber(null)).toBe(null);
+    })
+
+    it('should return number with correct precision', () => {
+      expect(osu.formatNumber(12.345, 2)).toBe('12.35');
+    })
+
+    it('shoudl return integer number', () => {
```

Comment:
`should return the number as an integer` or better describing the test, `formats the number without precision`